airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/io/S3DataLakeUtil.kt

```diff
@@ -273,6 +271,14 @@ class S3DataLakeUtil(
         // For example, this should allow us to use AWS instance profiles.
         val awsAccessKeyId = config.awsAccessKeyConfiguration.accessKeyId
         val awsSecretAccessKey = config.awsAccessKeyConfiguration.secretAccessKey
+
+        // Validate that either both credentials are provided or neither (for instance profiles)
+        if ((awsAccessKeyId == null) != (awsSecretAccessKey == null)) {
+            throw IllegalArgumentException(
+                "Both access_key_id and secret_access_key must be provided together, or neither (to use instance profiles)"
```

Comment:
in assume role mode, we always use the key+secret https://github.com/airbytehq/airbyte/blob/78c43755e47bc35c3ee64034f894c257993a0dbb/airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/GlueCredentialsProvider.kt#L58

Reply:
Done!

PTAL @edgao